crates/wash-lib/src/generate/git.rs

```diff
@@ -41,7 +41,16 @@ pub async fn clone_git_template(opts: CloneTemplate) -> Result<()> {
     };
 
     let cmd_out = Command::new("git")
-        .args(["clone", &repo_url, "--depth", "1", "--no-checkout", "."])
+        .args([
+            "clone",
+            &repo_url,
+            "--depth",
+            "1",
+            "--no-checkout",
+            "--branch",
+            &opts.repo_branch,
```

Comment:
Note that, if not specified, this will be `"main"`

Reply:
Thanks for noting this -- I see where the default gets set in `prepare_local_template` and it is weird to not be building this a bit more dynamically but we can cross that bridge when we get there